### PR TITLE
chore: update linters 2024

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-python/issues/2582

Recent VSCode versions were using more recent versions of Pyright which was showing errors. This updates our linter setup.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # See https://pre-commit.com/hooks.html for more hooks
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.5.0
+    rev: v5.0.0
```
Comment (dgozman) [inline .pre-commit-config.yaml:5]: What is this one?

File: local-requirements.txt [hunk 2]
```diff
@@ -7,7 +7,7 @@ mypy==1.11.2
 objgraph==3.6.1
 Pillow==10.4.0
 pixelmatch==0.3.0
-pre-commit==3.4.0
+pre-commit==3.5.0
```
Comment (dgozman) [inline local-requirements.txt:10]: What is this one? Why is it different from the previous file?

File: local-requirements.txt [hunk 3]
```diff
@@ -7,7 +7,7 @@ mypy==1.11.2
...
 pyOpenSSL==24.2.1
```
Comment (dgozman) [inline local-requirements.txt:11]: This one does not seem to match the previous file either.

File: pyproject.toml [hunk 4]
```diff
@@ -35,10 +35,12 @@ profile = "black"
...
 include = ["playwright", "tests", "scripts"]
+exclude = ["**/node_modules", "**/__pycache__", "**/.*", "./build"]
 pythonVersion = "3.8"
 reportMissingImports = false
 reportTypedDictNotRequiredAccess = false
 reportCallInDefaultInitializer = true
 reportOptionalSubscript = false
 reportUnboundVariable = false
 strictParameterNoneValue = false
+reportIncompatibleVariableOverride = false
```
Comment (dgozman) [inline pyproject.toml:46]: What is this one? Do we actually want to enable it?

File: setup.cfg [hunk 5]
```diff
@@ -3,3 +3,5 @@ ignore =
     E501
     W503
     E302
+    # Conflicts with black
```
Comment (dgozman) [inline setup.cfg:6]: More details would be beneficial.

**Review by dgozman — APPROVED** [review]
IIUC, this has no behavior change for the user, has it?

File: pyproject.toml (hunk 4 above)
Comment (mxschmitt) [reply pyproject.toml:46]: We override `self._loop` from being `Optional` to being not `Optional`. Mypy doesn't allow it anymore after they enabled this setting. Keeping it disabled seems fine for now imo.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mxschmitt) [reply .pre-commit-config.yaml:5]: Hooks which we consume from the `pre-commit` project - they are not shipped automatically so we specify them like that.

File: local-requirements.txt (hunk 2 above)
Comment (mxschmitt) [reply local-requirements.txt:10]: This is `pre-commit` - like ESLint - and above are some rules we consume.

File: local-requirements.txt (hunk 3 above)
Comment (mxschmitt) [reply local-requirements.txt:11]: In both cases the latest version. Seems like its like `definitely-typed` - so their version can differ.

**Comment by mxschmitt** [comment]
> IIUC, this has no behavior change for the user, has it?

no user facing change yes.

**Pull request merged**